### DISCUSSION TITLE
Is it possible to fine tune execution of lua filters in an extension that bundles several of them?

### DISCUSSION THREAD

**Discussion by ute** [quarto-dev/quarto-cli#8877] [Q&A]
### Description

The possibility to fine tune where in the AST processing a filter is inserted is fantastic! I am working on a set of filters that play together and should be inserted at different stages in the process, before and after quarto. Is it possible to specify this in the `_extension.yml`, so that the document only needs to specify application of the bundled extension? 
This would simplify the use very much, but I could not figure out how to do this - the extension should ideally work for both default and book project types. 

Edit: I realized that it is possible in principle with formats, but this precludes using the extension with other custom formats, or doesn't it?

**Comment by mcanouil** [comment]
The following should work even in extensions: <https://quarto.org/docs/prerelease/1.4/lua_changes.html#more-precise-targeting-of-ast-processing-phases>.

I have to say, that your use case is quite abstract to me, so unsure exactly what to answer.
Extensions can be filter(s), shortcode(s), format(s), project(s), or all of those things.

**Reply by multimeric** [reply]
Unfortunately the phase-targeting doesn't work in extensions, just in documents. See: https://github.com/quarto-dev/quarto-cli/discussions/11749.
